### PR TITLE
Allow advanced multi-value user data filter

### PR DESCRIPTION
**Pull request by millerdev**
https://trello.com/c/2FPRIvrQ/236-update-reminders-ui-to-allow-checking-a-property-against-multiple-values-in-special-user-data-filter

screenshot 2018-08-20 16 24 04

@gcapalbo

### CODE REVIEW

File: corehq/messaging/scheduling/forms.py [hunk 1]
```diff
@@ -2219,13 +2232,26 @@ def clean_user_data_property_name(self):
...
 
+        if use_user_data_filter == self.JSON:
+            err = _("Invalid JSON value. Expected a list of strings.")
+            try:
+                value = json.loads(value)
+            except Exception:
+                raise ValidationError(err)
+            if (not isinstance(value, list) or
+                    any(not isinstance(v, six.text_type) for v in value)):
+                raise ValidationError(err)
```
Comment (gcapalbo) [inline corehq/messaging/scheduling/forms.py]: could we also validate that it has at least one element in the list?

**Comment by gcapalbo** [comment]
@millerdev i've also been adding a label (see usage of `get_system_admin_label()` in the forms) next to all superuser features so that it's clear when making documentation which options are visible only to superusers and which are visible to all. could we add that label next to the value field when the `JSON` option is selected?

File: corehq/messaging/scheduling/forms.py (hunk 1 above)
Comment (millerdev) [reply corehq/messaging/scheduling/forms.py]: 593ff4d352cc1e3c8425d369dde0c860daa397dc

**Comment by millerdev** [comment]
screenshot 2018-08-21 08 45 38

**Review by gcapalbo — APPROVED** [review]

**Pull request merged**